src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -595,35 +609,23 @@ public boolean hasNodeFile() {
     }
 
     /**
-     * Returns all paths where lucene data will be stored, if a index.data_path
-     * setting is present, will return the custom data path to be used
-     */
-    public Path[] shardDataPaths(ShardId shardId, @IndexSettings Settings indexSettings) {
-        assert indexSettings != ImmutableSettings.EMPTY;
-        assert assertEnvIsLocked();
-        if (hasCustomDataPath(indexSettings)) {
-            return new Path[] {resolveCustomLocation(indexSettings, shardId)};
-        } else {
-            return shardPaths(shardId);
-        }
-    }
-
-    /**
      * Returns all shard paths excluding custom shard path
+     *
+     * @see #hasCustomDataPath(org.elasticsearch.common.settings.Settings)
+     * @see #resolveCustomLocation(org.elasticsearch.common.settings.Settings, org.elasticsearch.index.shard.ShardId)
+     *
      */
     public Path[] shardPaths(ShardId shardId) {
```

Comment:
I wonder if we should name this `shardPathOptions` or something that indicates the shard will be in one of these folder. At the very least I think we should add this to the java docs?